apps/backend/src/authn/github.strategy.ts

```diff
@@ -0,0 +1,57 @@
+import {Injectable, UnauthorizedException} from '@nestjs/common';
+import {PassportStrategy} from '@nestjs/passport';
+import axios from 'axios';
+import {Strategy} from 'passport-github';
+import {ConfigService} from '../config/config.service';
+import {AuthnService} from './authn.service';
+
+@Injectable()
+export class GithubStrategy extends PassportStrategy(Strategy, 'github') {
+  constructor(
+    private readonly authnService: AuthnService,
+    private readonly configService: ConfigService
+  ) {
+    super({
+      clientID: configService.get('GITHUB_CLIENTID') || 'disabled',
+      clientSecret: configService.get('GITHUB_CLIENTSECRET') || 'disabled',
+      scope: 'user:email',
+      passReqToCallback: true
+    });
+  }
+
+  async validate(
+    req: Record<string, unknown>,
+    accessToken: string
+  ): Promise<any> {
+    // Get user's linked emails from Github
+    const githubEmails = await axios.get('https://api.github.com/user/emails', {
+      headers: {Authorization: `token ${accessToken}`}
+    });
+    // Get user's info
+    const userInfoResponse = await axios.get('https://api.github.com/user', {
+      headers: {Authorization: `token ${accessToken}`}
+    });
+    const firstName: string = userInfoResponse.data.name.substr(
+      0,
+      userInfoResponse.data.name.indexOf(' ')
+    );
+    const lastName: string = userInfoResponse.data.name.substr(
+      userInfoResponse.data.name.indexOf(' ') + 1
+    );
```

Comment:
This code seems basically the same as the chunk below, can you just make a `splitName` function?